packages/cdktf/lib/runtime.ts

```diff
@@ -0,0 +1,49 @@
+// ----------------------------------------------------------------------
+// PROPERTY MAPPERS
+//
+// These are used while converting generated classes/property bags to Terraform JSON objects
+//
+// We use identity mappers for the primitive types. These don't do anything but are there to make the code
+// generation work out nicely (so the code generator doesn't need to emit different code for primitive
+// vs. complex types).
+export type Mapper = (x: any) => any;
+
+function identity(x: any) {
+  return x;
+}
+
+export const stringToTerraform: Mapper = identity;
+export const booleanToTerraform: Mapper = identity;
+export const anyToTerraform: Mapper = identity;
+export const numberToTerraform: Mapper = identity;
+
+export function listMapper(elementMapper: Mapper): Mapper {
+  return (x: any) => {
+    if (!canInspect(x)) { return x; }
+    return x.map(elementMapper);
+  };
+}
+
+export function hashMapper(elementMapper: Mapper): Mapper {
+  return (x: any) => {
+    if (!canInspect(x)) { return x; }
+
+    const ret: any = {};
+
+    Object.keys(x).forEach((key) => {
+        ret[key] = elementMapper(x[key]);
+    });
+
+    return ret;
+  };
+}
```

Comment:
The `runtime.ts` file doesn't need to be converted by jsii. It should only be needed by the jsii-runtime.
I might need to reflow `synthesizeAttributes` some to make it realize that. Or maybe there is some other way to tell jsii to ignore it.

Reply:
I think that's why [this file](https://github.com/hashicorp/terraform-cdk/blob/51ef72c701d93b382d7c39a12fd6cd47b558a2bb/packages/cdktf/lib/private/unique.ts) is in a `private` folder.

Reply:
Just putting it in a `private` folder isn't enough. Since external code needs to reference it, it ends up needing to exist in the index.

Some options:

- Just ignore them (they'd still print unless we ignored the category)
- Reach out to the jsii team to see if there is a way to ignore a file
- Rebuild the functions into a class that is jsii compliant
- Build into a separate library that isn't ran through jsii at all and reference it.

It would be nice if we could just ignore, but refactoring probably won't be too bad.

Reply:
> * Reach out to the jsii team to see if there is a way to ignore a file

Perhaps [this](https://github.com/aws/jsii/blob/818ce231ec78b73a2a13f86f751b5aa2ba0a8b68/docs/configuration.md#excludetypescript) is what we're looking for here?

Reply:
No luck with that. Opened [an issue](https://github.com/aws/jsii/issues/2085) to see if I've missed something.

Reply:
I heard back from the jsii team and there isn't currently way to ignore.

My preference is to just not worry about the warnings right now, but I can look into refactoring them to be jsii compliant.

Reply:
@skorfmann thoughts?

Reply:
👍